src/lib/rest/StringFilter.cpp

```diff
@@ -1307,7 +1671,18 @@ bool StringFilter::match(ContextElementResponse* cerP)
       break;
 
     case SfopMatchPattern:
-      return itemP->matchPattern(caP);
+      //
+      // To not be forgotten for CNR (and discussed during PR-review):
```

Comment:
By your comprehensive description, I agree this is a bugfix and should be mentioned in the CNR (and probably covered by a .test).

(Side-note: it seem this bug was introduced by me when I implemented `~=` filter (https://github.com/telefonicaid/fiware-orion/blame/develop/src/lib/rest/StringFilter.cpp#L1310). I didn't realized of the "return pattern" in above cases and trying to achieve something more compact I introduced the bug).